### ISSUE TITLE
Chatlist CSRF token going stale.

### ISSUE THREAD

**Issue by klondikemarlen** [elthran/UndyingKingdoms#183]
This happens when you leave the chat window open for too long (as I am wont to do).

The chat tries to update but can't because the CSRF token is too old. The only solution is to do a hard refresh of the page. I'm not sure why a soft refresh doesn't work; as far as I can tell it should.

I'm honestly not sure how to fix this bug? Unless the chat automatically updated at least once every hour (or right before chat token goes stale). Or perhaps the chat could do a query for a new token right before each request? ... except in needs a valid token to query ..

**Comment by klondikemarlen** [comment]
I think I maybe I need to switch the chat first load to the old method using jinaj instead of the new method where data gets pulled as json ...

**Comment by klondikemarlen** [comment]
This one really bugs me but I still have no idea how to fix it. I did some research and it looks like WTForms has some kind of extra CSRF token generation magic, but I'm still not sure how to access that on the frontend. I wish I could generate CSRF client side.

**Comment by klondikemarlen** [comment]
```html
<META HTTP-EQUIV="REFRESH" CONTENT="csrf_timeout_in_seconds">
```

from https://stackoverflow.com/a/27501327

**Comment by klondikemarlen** [comment]
Fixed in 61ea47812ec5b35723a0742a9c9320d1c45f5217
